tests/test_inference.py

```diff
@@ -1,18 +1,135 @@
+import os
+import random
+
+import pytest
+import torch
+
 from cities.modeling.model_interactions import InteractionsModel
+from cities.queries.causal_insight import CausalInsight
+from cities.utils.cleaning_utils import find_repo_root
+from cities.utils.data_grabber import list_interventions, list_outcomes
+
+os.environ["CUDA_VISIBLE_DEVICES"] = ""
+device = torch.device("cpu")
+
+root = find_repo_root()
+
+interventions = list_interventions()
+outcomes = list_outcomes()
+shifts = [1, 2, 3]
 
+# TODO add list_fips() function
+# and parametrize with random fips
+# and uniformly random intervention value
+fips = 1003
+intervened_value = 0.9
 
-# TODO this needs to be parametrized by outcome datasets,
-# intervention datasets and forward shifts
-def test_InteractionsModel():
+# running for all is too inefficient
+# let's just run for random on a regular basis
+# comment out for a full battery of tests
+# intervention = [random.choice(interventions)]
+# outcome = [random.choice(outcomes)]
+
+interventions = [random.choice(interventions)]
+outcomes = [random.choice(outcomes)]
+
+
+# forward shifts caused some shape troubles,
+# so we test for all of them at each random test
+@pytest.mark.parametrize("intervention", interventions)
+@pytest.mark.parametrize("outcome", outcomes)
+@pytest.mark.parametrize("shift", shifts)
+def test_InteractionsModel(intervention, outcome, shift):
```

Comment:
Maybe call this `test_smoke_interactionsmodel` so it's more clear that it's not actually testing correctness, just that it runs.

Reply:
This comment applies to most of the below as well.

Reply:
done